File: jpa-separate-ro-rw/src/test/java/com/sight/testType/BaseTest.java
package com.sight.testType;

import com.sight.config.AppProfile;
import org.springframework.test.context.ActiveProfiles;

import java.nio.file.Path;
import java.nio.file.Paths;

@ActiveProfiles(AppProfile.LOCAL)
public interface BaseTest {

    default String getResourcePath() {
        Path resourcePath = Paths.get("src", "test", "resources");
        return resourcePath.toFile().getAbsolutePath();
    }
}

File: JPAwithRecursive/src/test/kotlin/com/sight/blog/config/test/BaseTest.kt
package com.sight.blog.config.test

import java.nio.file.Path
import java.nio.file.Paths

interface BaseTest {

    fun getResourcePath(): String {
        val resourceDirectory: Path = Paths.get("src", "test", "resources")
        return resourceDirectory.toFile().absolutePath
    }
}